src/p_misc.c

```diff
@@ -213,151 +393,282 @@ prim_fmttime(PRIM_PROTOTYPE)
 #endif
 }
 
+/**
+ * Implementation of MUF USERLOG
+ *
+ * Consumes a string, and formats it into a message to write to the user log,
+ * which is by default 'logs/user'.  Requires MUCKER level tp_userlog_mlev.
+ *
+ * Format is:
+ * player(#) [program.muf(#)] timestamp: message
+ *
+ * @see log_user
+ *
+ * @param player the player running the MUF program
+ * @param program the program being run
+ * @param mlev the effective MUCKER level
+ * @param pc the program counter pointer
+ * @param arg the argument stack
+ * @param top the top-most item of the stack
+ * @param fr the program frame
+ */
 void
 prim_userlog(PRIM_PROTOTYPE)
 {
     CHECKOP(1);
     oper1 = POP();
+
     if (oper1->type != PROG_STRING)
-	abort_interp("Non-string argument.");
+        abort_interp("Non-string argument.");
+
     if (mlev < tp_userlog_mlev)
-	abort_interp("Permission Denied (mlev < tp_userlog_mlev)");
+        abort_interp("Permission Denied (mlev < tp_userlog_mlev)");
+
+    /**
+     * @TODO buf = DoNullInd(oper1->data.string);
```

Comment:
This is a pretty generic @TODO ... is the purpose to add the line that you have here?  Is there something else to do here?

Reply:
Changed to say that it may be able to replace the "if" construct entirely.